openlibrary/templates/books/edit/edition.html

```diff
@@ -649,78 +651,3 @@
             </div>
         </div>
     </fieldset>
-
-<script type="text/javascript">
-<!--
-    window.q.push(function() {
-        // FIXME: Error code duplicated in various templates.
-        // Should be pushed into a macro.
-        function error(errordiv, input, message) {
-            \$(errordiv).show().html(message);
-            \$(input).focus();
-            return false;
-        }
-
-        \$("#roles").repeat({
```

Comment:
Let's make this similar to how we did the previous PR, so add a `class` and `data-config` to each of these (ie to the element with id `roles`, `identifiers`, and `classifications`). Maybe something like eg `class="identifiers-table" data-preset="roles"`. That makes them behave more like components and will make it easier for us to replace them with components later on.

Note I think it's easier if these all get the same class, `identifiers-table`, and then store the "id" somewhere (ie `data-preset`). In the other PR we ended with a separate JS file for every `multi-input-autocomplete`, which is not good for performance. If we use a single class for all of these, then there will only be one JS file for these 3 and one entry in index.js .

Note in the long term, these 3 are the things that will likely be replaced by your AuthorsIdentifiers component :)

Reply:
Looks resolved to me!